shared/naturalcrit/markdown.js

```diff
@@ -237,32 +237,45 @@ const superSubScripts = {
 const definitionLists = {
 	name  : 'definitionLists',
 	level : 'block',
-	start(src) { return src.match(/^.*?::.*/m)?.index; },  // Hint to Marked.js to stop and check for a match
+	start(src) { return src.match(/^.*?::.*\n\n/m)?.index; },  // Hint to Marked.js to stop and check for a match
 	tokenizer(src, tokens) {
-		const regex = /^([^\n]*?)::([^\n]*)(?:\n|$)/ym;
+		const regex = /^([^\n:]*?)::(.*)(?:\n|$)/ym;
 		let match;
-		let endIndex = 0;
-		const definitions = [];
+		const endIndex = src.match(`\n\n`)?.index + 2;
+		const allDefinitions = [];
+		let currentDefinition = [];
 		while (match = regex.exec(src)) {
-			definitions.push({
-				dt : this.lexer.inlineTokens(match[1].trim()),
-				dd : this.lexer.inlineTokens(match[2].trim())
-			});
-			endIndex = regex.lastIndex;
+			if(match[1].trim()?.length) {
+				if(currentDefinition?.dt?.length) {
+					allDefinitions.push(currentDefinition);
+					currentDefinition = [];
+				}
+				currentDefinition = {
+					dt : this.lexer.inlineTokens(match[1].trim()),
+					dd : []
+				};
+			}
+			currentDefinition.dd = currentDefinition.dd.concat(match[2].split('::').filter((item)=>item).map((s)=>this.lexer.inlineTokens(s.trim())));
+			if(!currentDefinition.dd?.length) {
+				currentDefinition.dd = [this.lexer.inlineTokens('')];
+			}
 		}
-		if(definitions.length) {
+		if(currentDefinition.hasOwnProperty('dt')) { allDefinitions.push(currentDefinition); }
+		if(allDefinitions.length) {
 			return {
-				type : 'definitionLists',
-				raw  : src.slice(0, endIndex),
-				definitions
+				type        : 'definitionLists',
+				raw         : src.slice(0, endIndex),
+				definitions : allDefinitions
 			};
 		}
 	},
 	renderer(token) {
-		return `<dl>${token.definitions.reduce((html, def)=>{
-			return `${html}<dt>${this.parser.parseInline(def.dt)}</dt>`
-			            + `<dd>${this.parser.parseInline(def.dd)}</dd>\n`;
-		}, '')}</dl>`;
+		let returnVal = `<dl>`;
+		token.definitions.forEach((def)=>{
+			const dds = def.dd.map((s)=>`<dd>${this.parser.parseInline(s)}</dd>`).join('\n');
+			returnVal += `<div><dt>${this.parser.parseInline(def.dt)}</dt>${dds}</div>`;
```

Comment:
The `<dt>` should not be inside a `<div>` here. Can you remove that?

Reply:
It is valid per spec and was recommended in some examples, but I have removed it.